File: benchmark/sieve.py
def sieve(n):
    count = 0
    flags = [True] * (n + 1)
    
    for i in range(2, n + 1):
        if flags[i]:
            count += 1
            j = i * 2
            while j <= n:
                flags[j] = False
                j += i
    return count

print("Sieving up to 1,000,000...")
print(sieve(1000000))

File: benchmark/sieve.lua
function sieve(n)
    local count = 0
    local flags = {}
    for i = 0, n do flags[i] = true end
    
    for i = 2, n do
        if flags[i] then
            count = count + 1
            local j = i * 2
            while j <= n do
                flags[j] = false
                j = j + i
            end
        end
    end
    return count
end

print("Sieving up to 1,000,000...")
print(sieve(1000000))
